apiserver/facades/agent/secretsmanager/register.go

```diff
@@ -62,14 +63,17 @@ func NewSecretManagerAPI(stdCtx context.Context, ctx facade.ModelContext) (*Secr
 	cloudService := serviceFactory.Cloud()
 	credentialSerivce := serviceFactory.Credential()
 
+	backendService := serviceFactory.SecretBackend(model.ControllerUUID(), provider.Provider)
 	secretBackendAdminConfigGetter := func(stdCtx context.Context) (*provider.ModelBackendConfigInfo, error) {
-		return secrets.AdminBackendConfigInfo(stdCtx, secrets.SecretsModel(model), cloudService, credentialSerivce)
+		return backendService.GetSecretBackendConfigForAdmin(stdCtx, coremodel.UUID(model.UUID()))
 	}
 	secretService := serviceFactory.Secret(secretBackendAdminConfigGetter)
```

Comment:
Ultimately, we should be passing in a getter which returns a backend for a given backend id.
When we process an external secret, we have the backend id. The mechanics of realising a backend from that should not need to be exposed to the secret service. We do that now but it should change.

Reply:
This is for getting backend config purposes.
If the secret service wants to get a backend provider directly by backend ID.
We can add that for secret stuff in the future.